samples/sample_encode/src/sample_encode.cpp

```diff
@@ -365,12 +365,37 @@ mfxStatus ParseInputString(msdk_char* strInput[], mfxU8 nArgNum, sInputParams* p
         }
         else if (0 == msdk_strcmp(strInput[i], MSDK_STRING("-la")))
         {
-            pParams->nRateControlMethod = MFX_RATECONTROL_LA;
+            if (!pParams->nRateControlMethod)
+            {
+                pParams->nRateControlMethod = MFX_RATECONTROL_LA;
+            }
+            else if (pParams->nRateControlMethod == MFX_RATECONTROL_ICQ)
+            {
+                pParams->nRateControlMethod = MFX_RATECONTROL_LA_ICQ;
+            }
+            else
+            {
+                PrintHelp(strInput[0], MSDK_STRING("More than one BRC modes assigned, and another BRC mode isn't compatible with LA."));
+                return MFX_ERR_UNSUPPORTED;
+            }
         }
         else if (0 == msdk_strcmp(strInput[i], MSDK_STRING("-lad")))
         {
             VAL_CHECK(i+1 >= nArgNum, i, strInput[i]);
-            pParams->nRateControlMethod = MFX_RATECONTROL_LA;
+            if (!pParams->nRateControlMethod)
+            {
+                pParams->nRateControlMethod = MFX_RATECONTROL_LA;
+            }
+            else if (pParams->nRateControlMethod == MFX_RATECONTROL_ICQ)
+            {
+                pParams->nRateControlMethod = MFX_RATECONTROL_LA_ICQ;
+            }
+            else
+            {
+                PrintHelp(strInput[0], MSDK_STRING("More than one BRC modes assigned, and another BRC mode isn't compatible with LA."));
```

Comment:
LA tests failed after your recent modifications. That's probably because you don't handle -la and -lad options in the same command line. I.e. once parser will meet -la it will set RC=LA and when it will meet -lad it will simply error out here.

Question you probably need to answer from the code: can -lad option present in the command line without -la? If it can't then remove processing of RC from here at all. If it can, then you need to add handling for LA and LA_ICQ modes in both branches.